pyrobosim/pyrobosim/core/yaml_utils.py

```diff
@@ -127,90 +126,37 @@ def add_robots(self):
-                self.world,
-                max_nodes=max_nodes,
-                max_connection_dist=max_connection_dist,
-            )
-
-        elif planner_type == "astar_grid":
-            from pyrobosim.navigation.astar_grid import AStarGridPlanner
-
-            resolution = planner_data.get("resolution", 0.05)
-            inflation_radius = planner_data.get("inflation_radius", 0.1)
-            heuristic = planner_data.get("heuristic", "euclidean")
-            diagonal_motion = planner_data.get("diagonal_motion", True)
-            max_time = planner_data.get("max_time", 5.0)
-            compress_path = planner_data.get("compress_path", True)
-
-            self.world.path_planner = AStarGridPlanner(
-                self.world,
-                resolution=resolution,
-                inflation_radius=inflation_radius,
-                diagonal_motion=diagonal_motion,
-                heuristic=heuristic,
-                compress_path=compress_path,
-                max_time=max_time,
-            )
-        else:
-            warnings.warn(f"Invalid global planner type specified: {planner_type}")
-
-        # Always make a search graph as we use it for other things.
-        max_edge_dist = planner_data.get("max_edge_dist", np.inf)
-        collision_check_dist = planner_data.get("collision_check_dist", 0.1)
-        self.world.create_search_graph(
-            max_edge_dist=max_edge_dist,
-            collision_check_dist=collision_check_dist,
-            create_planner=create_planner,
-        )
-
     def get_local_path_planner(self, robot_data):
         """Gets local planner path planner to a robot."""
         if "path_planner" not in robot_data:
             return None
+
+        from pyrobosim.navigation import PathPlanner
+
         planner_data = robot_data["path_planner"]
         planner_type = planner_data["type"]
 
         if planner_type == "rrt":
-            from pyrobosim.navigation.rrt import RRTPlanner
-
             bidirectional = planner_data.get("bidirectional", False)
             rrt_star = planner_data.get("rrt_star", False)
             rrt_connect = planner_data.get("rrt_connect", False)
             max_connection_dist = planner_data.get("max_connection_dist", 0.5)
             max_nodes_sampled = planner_data.get("max_nodes_sampled", 1000)
             max_time = planner_data.get("max_time", 5.0)
             rewire_radius = planner_data.get("rewire_radius", 1.0)
-            return RRTPlanner(
-                self.world,
-                bidirectional=bidirectional,
-                rrt_star=rrt_star,
-                rrt_connect=rrt_connect,
-                max_connection_dist=max_connection_dist,
-                max_nodes_sampled=max_nodes_sampled,
-                max_time=max_time,
-                rewire_radius=rewire_radius,
-            )
+            planner_config = {
+                "grid": None,  # TODO : How should a grid be added based on yaml config
```

Comment:
I dont know if we need a `OccupancyGrid.fromYaml()`.
Since the yaml info will be read anyway when loading the robot's planner, we can just call the `occupancy_grid_from_world()` there and pass in the grid to the planner.